src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  EuiButton,
+  EuiFormRow,
+  EuiText,
+  EuiSpacer,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonIcon,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { SavedObjectsStart } from '../../../../../core/public';
+import { getDataSourcesList } from '../../utils';
+import { DataSource } from '../../../common/types';
+
+export interface SelectDataSourcePanelProps {
+  errors?: { [key: number]: string };
+  savedObjects: SavedObjectsStart;
+  selectedDataSources: DataSource[];
+  onChange: (value: DataSource[]) => void;
+}
+
+export const SelectDataSourcePanel = ({
+  errors,
+  onChange,
+  selectedDataSources,
+  savedObjects,
+}: SelectDataSourcePanelProps) => {
+  const [dataSourcesOptions, setDataSourcesOptions] = useState<EuiComboBoxOptionOption[]>([]);
+  useEffect(() => {
+    if (!savedObjects) return;
+    getDataSourcesList(savedObjects.client, ['*']).then((result) => {
```

Comment:
As we use `*` here, what about workspace owner go to the edit page, will it throw error and prevent the owner to update other workspace attributes like name/description?

Reply:
There will be a new PR later to disable data source assignment.